com.unity.ml-agents/Runtime/Sensors/RayPerceptionSensor.cs

```diff
@@ -279,27 +290,58 @@ public RayPerceptionOutput RayPerceptionOutput
             get { return m_RayPerceptionOutput; }
         }
 
-        void SetNumObservations(int numObservations)
+        /// <summary>
+        /// The observation size per ray.
+        /// Override this method for custom observations.
+        /// </summary>
+        public virtual int GetObservationSizePerRay()
+        {
+            return RayPerceptionInput.DetectableTags.Count + 2;
+        }
+
+        /// <summary>
+        /// The number of rays in the sensor.
+        /// Override this method for custom observations.
+        /// </summary>
+        public int GetNumberOfRays()
```

Comment:
Needs to be `virtual` if it's meant to be overridden.

Reply:
I don't think this one can be overridden since we handle the angle calculations internally. I added this as more like a helper method.

Reply:
Makes sense, but in that case, you need to remove "Override this method..." from the docstring :)

Reply:
got copy-pasted. now removed